plutus-ledger-constraints/src/Ledger/Constraints/OnChain.hs

```diff
@@ -107,7 +108,7 @@ checkTxConstraint ctx@ScriptContext{scriptContextTxInfo} = \case
         $ V.findDatum dvh scriptContextTxInfo == Just dv
     MustSatisfyAnyOf xs ->
         traceIfFalse "Ld" -- "MustSatisfyAnyOf"
-        $ any (checkTxConstraint ctx) xs
+        $ or $ map (all (checkTxConstraint ctx)) xs
```

Comment:
That's just `any (all (checkTxConstraint ctx)) xs` isn't it?

Reply:
Ahah, yeah, missed that, thanks!